ddtrace/api.py

```diff
@@ -146,23 +146,50 @@ def _downgrade(self):
         self._set_version(self._fallback)
 
     def send_traces(self, traces):
-        if not traces:
-            return
+        """Send traces to the API.
 
+        :param traces: A list of traces.
+        :return: The list of API HTTP responses.
+        """
         start = time.time()
+        responses = []
         payload = Payload(encoder=self._encoder)
         for trace in traces:
-            payload.add_trace(trace)
+            try:
+                payload.add_trace(trace)
+            except PayloadFull:
+                # Is payload full or is the trace too big?
+                # If payload is not empty, then using a new Payload might allow us to fit the trace.
+                # Let's flush the Payload and try to put the trace in a new empty Payload.
+                if not payload.empty:
+                    responses.append(self._flush(payload))
```

Comment:
`_flush` could raise an exception, how should we handle this?

Reply:
I assume we want to catch them and try all of the other potential payloads first.

we could either get a connection issue, or an error response from the agent (typically if we are rate limited by the agent, payload too large, invalid request body, etc).

Reply:
(Just to be clear, this is already a problem with the current code, right?)

Reply:
sort of. an exception being raised is fine, because it is handled by the writer/worker, but now that we are potentially making multiple API calls, seems like we should do something in here for that?

e.g. if we break into 3 payloads and the first one fails, we should definitely still attempt the other 2?

Reply:
I'd say yes. I'll add something for that.